File: back-end/src/main/java/com/action/entity/NewsHeadline.java
package com.action.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.io.Serializable;
import java.util.Date;

/**
 * 新闻头条实体类
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
public class NewsHeadline implements Serializable {
    private Integer hid;
    
    private String title;
    
    private String article;
    
    private Integer type;
    
    private Integer publisher;
    
    private Integer pageViews;
    
    private Date createTime;
    
    private Date updateTime;
    
    private Integer isDeleted;

}

File: back-end-gin/models/news_headline.go
package models

import "time"

// NewsHeadline 新闻头条实体类
type NewsHeadline struct {
	Hid        int       `gorm:"column:hid;primaryKey;autoIncrement" json:"hid"`
	Title      string    `gorm:"column:title;type:varchar(200);not null" json:"title"`
	Article    string    `gorm:"column:article;type:text" json:"article"`
	Type       int       `gorm:"column:type;not null" json:"type"`
	Publisher  int       `gorm:"column:publisher;not null" json:"publisher"`
	PageViews  int       `gorm:"column:page_views;default:0" json:"pageViews"`
	CreateTime time.Time `gorm:"column:create_time;autoCreateTime" json:"createTime"`
	UpdateTime time.Time `gorm:"column:update_time;autoUpdateTime" json:"updateTime"`
	IsDeleted  int       `gorm:"column:is_deleted;default:0" json:"isDeleted"`
}

// TableName 指定表名
func (NewsHeadline) TableName() string {
	return "news_headline"
}

